pkg/mixin/query/query.go

```diff
@@ -46,39 +46,51 @@ type MixinInputGenerator interface {
 	BuildInput(mixinName string) ([]byte, error)
 }
 
+type MixinBuildOutput struct {
+	// Name of the mixin.
+	Name string
+
+	// Stdout is the contents of stdout from calling the mixin.
+	Stdout string
+
+	// Error returned when the mixin was called.
+	Error error
+}
+
 // Execute the specified command using an input generator.
 // For example, the ManifestGenerator will iterate over the mixins in a manifest and send
 // them their config and the steps associated with their mixin.
-func (q *MixinQuery) Execute(ctx context.Context, cmd string, inputGenerator MixinInputGenerator) (map[string]string, error) {
+// The mixins are queried in parallel but the results are sorted in the order that the mixins were defined in the manifest.
+func (q *MixinQuery) Execute(ctx context.Context, cmd string, inputGenerator MixinInputGenerator) ([]MixinBuildOutput, error) {
 	ctx, span := tracing.StartSpan(ctx)
 	defer span.EndSpan()
 
 	mixinNames := inputGenerator.ListMixins()
-	results := make(map[string]string, len(mixinNames))
-	type queryResponse struct {
-		mixinName string
-		output    string
-		runErr    error
-	}
+	results := make([]MixinBuildOutput, len(mixinNames))
+	resultLookup := make(map[string]int, len(mixinNames))
 
-	var responses = make(chan queryResponse, len(mixinNames))
 	gerr := errgroup.Group{}
 
-	for _, mn := range mixinNames {
-		mixinName := mn // Force mixinName to be in the go routine's closure below
+	for i, mn := range mixinNames {
+		// Force variables to be in the go routine's closure below
+		i := i
+		mn := mn
+		results[i].Name = mn
+		resultLookup[mn] = i
```

Comment:
This looks unused, it is here for future capabilities to lookup the index of a mixin name?

Reply:
Oops! I thought I needed it but then after making some more changes, you're right it's not used anymore. I'll remove it.